scripts/__input_class_player/__input_class_player.gml

```diff
@@ -1019,6 +1030,49 @@ function __input_class_player() constructor
         }
     }
     
+    static __trigger_effect_set = function(_trigger, _effect, _set)
+    {
+        var _gamepad = __source_get_gamepad();
+        if ((_gamepad < 0) || !is_struct(_effect)) return;
+        
+        if (__trigger_effect_paused)
+        {
+            __input_trace("Warning! New trigger effect ignored, player ", __index, " trigger effect is paused");
+            return;
+        }
+
+        var _intercepted = !global.__input_gamepads[_gamepad].__trigger_effect_apply(_trigger, _effect);
+        
+        if (!_set) return;
+        if (_trigger == gp_shoulderlb)
+        {
+            __trigger_intercepted_left = _intercepted;
+            __trigger_effect_left      = _effect;
+        }
+        else
+        {
+            __trigger_intercepted_right = _intercepted;
+            __trigger_effect_right      = _effect;                 
+        }
+    }
+    
+    static __trigger_effect_pause = function(_state)
+    {   
+        __trigger_effect_paused = _state;
+    
+        if (!_state)
+        {
+            __input_player_apply_trigger_effects(__index);
+        }
+        else
+        {
+            var _gamepad = __source_get_gamepad();
```

Comment:
How does this interact with multidevice mode?

Reply:
First pad only. I suppose mixed and multi can apply for all used sources. The haptic effects do a little “click” when you change modes but I suppose this is more tolerable than “vibrate everything” since they don’t actuate without trigger action.